src/output/outputMod.F90

```diff
@@ -1782,17 +1823,94 @@ subroutine storeSurfsolInBuffer(sps, buffer, nn, blockID,   &
-
        fact = two/(gammaInf*pInf*MachCoef*MachCoef)
 
        do j=rangeFace(2,1), rangeFace(2,2)
           do i=rangeFace(1,1), rangeFace(1,2)
              nn = nn + 1
-             buffer(nn) = fact*(half*(pp1(i,j) + pp2(i,j)) - pInf)
-          enddo
+             ! mham Cp normalisation:
+             rrate_=cgnsdoms(1)%rotrate
+             r_(1) =   (half*(xx1(i,j,1) + xx2(i,j,1)))
+             r_(2) =   (half*(xx1(i,j,2) + xx2(i,j,2)))
+             r_(3) =   (half*(xx1(i,j,3) + xx2(i,j,3)))
+             ! calc cross-product between rotation rate and r_
+             wCrossR(1) = rrate_(2)*r_(3) - rrate_(3)*r_(2)
+             wCrossR(2) = rrate_(3)*r_(1) - rrate_(1)*r_(3)
+             wCrossR(3) = rrate_(1)*r_(2) - rrate_(2)*r_(1)
+             rot_speed = SQRT(wCrossR(1)**2 +wCrossR(2)**2 +wCrossR(3)**2 )
+             buffer(nn) = ((half*(pp1(i,j) + pp2(i,j)) - pInf)*pRef) &
+                  / (half*(rhoInfDim)*((uInf*uRef)**2 + (rot_speed)**2 ))
+             ! mham: Comments on the computations above - no code 
+             ! Note, that we take rrate_(1) since we rotate 
+             !
+             ! Cp = (P_i - P_0) / (0.5*rho*(U_a)^2)
+             !
+             ! Nominator: (P_i-P_0) -> (half*(pp1Dim(i,j)+pp2Dim(i,j))-pInfDim)
+             ! pp1Dim  = pp1 * pRef
+             ! pp2Dim  = pp2 * pRef
+             ! pInfDim = pInf * pRef             
+             ! mham: pRef is defined in  module flowVarRefState (see the 'save')
+             !       but pRef is first set in the subr referenceState. However,
+             !       since it is defined in module flowVarRefState and we load
+             !       that module above it should be available to us.
+             !
+             ! Denominator: (0.5*rho*(U_a)^2) -> 
+             !       factDIM = two/(rhoDim * ((V_infDim)**2) + ((rDim*wDim)**2))
+             ! rhoDim  = rhoInfDim ! l. 81, initializeFlow.F90
+             ! V_infDim   = uInf*uRef ! also from initializeFlow.F90, l. 83
+          enddo
+          ! mham: Comments on the computations above - no code 
+          ! mham: pp1 and pp2 are just pressure-pointers, e.g.
+          ! pp1    => p(1,1:,1:);
+          ! Thus, they point to p and should be rescaled back to SI
+          ! mham: as explained in the opening paragraph of this subroutine, we
+          ! have to get the average of the p in the first cell and the p in the
+          ! ghostcell/halo. This explains: half*(pp1(i,j) + pp2(i,j))
+          ! From start of subr:
+          !       As the solution
+          !       must be stored in the center of the boundary face the average
+          !       value of the first internal cell and its corresponding halo is
+          !       computed. The counter nn is updated in this routine. However
+          !       it is not initialized, because multiple contributions may be
+          !       stored in buffer.
        enddo
+       ! mham: Comments on the computations above - no code 
```

Comment:
This looks very good to have here imo.
(Gotta love a comment block that includes "as explained in the opening paragraph of this subroutine")